pandas/core/generic.py

```diff
@@ -629,6 +631,103 @@ def _set_axis(self, axis, labels):
         self._data.set_axis(axis, labels)
         self._clear_item_cache()
 
+    def set_index(self, keys, drop=True, append=False, inplace=False,
+                  verify_integrity=False):
+        """
+        Set the index (row labels) using one or more given arrays (or labels).
+
+        Parameters
+        ----------
+        %(params)s
+        append : bool, default False
+            Whether to append columns to existing index.
+        inplace : bool, default False
+            Modify the %(klass)s in place (do not create a new object).
+        verify_integrity : bool, default False
+            Check the new index for duplicates. Otherwise defer the check until
+            necessary. Setting to False will improve the performance of this
+            method.
+
+        Returns
+        -------
+        %(klass)s
+            The reindexed %(klass)s. Will be None if inplace is True.
+
+        See Also
+        --------
+        %(other_klass)s.set_index: Method adapted for %(other_klass)s.
```

Comment:
I think your validation error is that there is no space before the colon on this line

Reply:
@WillAyd 
If that would cause the error then the `[Series|DataFrame].set_index` docstrings should fail too, shouldn't they? Locally they pass, but let's hope you're right. :)

Reply:
Hmm yea. So it does fail locally on NDFrame.set_index with the same error as shown in Azure logs